src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,50 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserNotOkException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_CHARS = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        try {
+            checkingUser(user);
+            storageDao.add(user);
+            return storageDao.get(user.getLogin());
+        } catch (UserNotOkException e) {
+            throw new RuntimeException(e);
+        }
     }
+
+    private boolean checkingUser(User user)
+            throws UserNotOkException {
+        if (user == null) {
+            throw new UserNotOkException("User is null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new UserNotOkException("Login is already taken.");
+        }
+
+        if (user.getLogin().length() < MIN_CHARS) {
```

Comment:
There is a potential `NullPointerException` if `user.getLogin()` is null. According to the checklist, don't use `NullPointerException`. Instead, add a separate check for `user.getLogin() == null` and throw a `UserNotOkException` with an appropriate message.